### PR TITLE
feat(context-based-restrictions): add service sdk, tests, and examples

### PR DESCRIPTION
**Pull request by agiammar**
## PR summary
Add context-based-restrictions go sdk with tests, integration tests, and examples

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
(added new service name link in README.md)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+)
+
+//
+// This file provides an example of how to use the Context Based Restrictions service.
+//
+// The following configuration properties are assumed to be defined:
+// CONTEXT_BASED_RESTRICTIONS_URL=<service base url>
+// CONTEXT_BASED_RESTRICTIONS_AUTH_TYPE=iam
+// CONTEXT_BASED_RESTRICTIONS_APIKEY=<IAM apikey>
+// CONTEXT_BASED_RESTRICTIONS_AUTH_URL=<IAM token service base URL - omit this if using the production environment>
```
Comment (padamstx) [inline contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go:40]: Please add comments for the `TEST_ACCOUNT_ID` and `TEST_SERVICE_NAME` properties which the code is also retrieving.

**Comment by Andras-Csanyi** [comment]
@agiammar I'd like to ask you to attach a screenshot about the green integration test results. Thank you!

**Review by padamstx — APPROVED** [review]
LGTM! Thank you for adding the negative tests in the integration test file!
I suggested one minor change to the examples code to document the additional properties used by the examples, but I'll approve now assuming that is addressed.

**Comment by agiammar** [comment]
Hi Andras,
thanks for the review!

I attached below the screenshot of the final report for the entire suite of tests (unit, integration, and examples). The tests were executed against CBR prod.

Screen Shot 2021-09-28 at 10 22 05 AM

File: contextbasedrestrictionsv1/context_based_restrictions_v1_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,1015 @@
...
+		})
+		It(`GetZone(getZoneOptions *GetZoneOptions) with 'missing required ZoneID parameter' error`, func() {
+			getZoneOptions := &contextbasedrestrictionsv1.GetZoneOptions{
+				TransactionID: getTransactionID(),
+			}
+
+			zone, response, err := contextBasedRestrictionsService.GetZone(getZoneOptions)
+
+			Expect(err).To(Not(BeNil()))
+			Expect(err.Error()).To(ContainSubstring("Field validation for 'ZoneID' failed"))
```
Comment (Andras-Csanyi) [inline contextbasedrestrictionsv1/context_based_restrictions_v1_integration_test.go:289]: I'd like to request a check for status code here.

File: contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go (hunk 1 above)
Comment (agiammar) [reply contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go:40]: Hi Phil,
thanks for the review!
I added the documentation about the additional config properties.

File: contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+			Expect(zone).ToNot(BeNil())
+			zoneID = *zone.ID
+
+		})
+		It(`ListZones request example`, func() {
+			fmt.Println("\nListZones() result:")
+			// begin-list_zones
+
+			listZonesOptions := contextBasedRestrictionsService.NewListZonesOptions(
+				testAccountID,
```
Comment (Andras-Csanyi) [inline contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go:159]: I'd like to ask you to rename this variable to `accountId` if it doesn't conflict. The point is not displaying `test` in the reference.

**Review by Andras-Csanyi — CHANGES_REQUESTED** [review]
Overall looks very good! I made a single change request.

Thank you very much for the effort you put into it. Especially, the negative test cases.
Here comes the point when I have to apologise for being very picky in this review. :)

File: contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go (hunk 3 above)
Comment (agiammar) [reply contextbasedrestrictionsv1/context_based_restrictions_v1_examples_test.go]: OK. No conflicts.

**Review by Andras-Csanyi — APPROVED** [review]
LGTM!

**Pull request merged**